typescript/sdk/src/token/checker.ts

```diff
@@ -86,37 +111,119 @@ export class HypERC20Checker extends ProxiedRouterChecker<
+      }
+    }
+  }
+
+  private cachedAllActualDecimals: Record<ChainName, number> | undefined =
+    undefined;
+
+  async getEvmActualDecimals(): Promise<Record<ChainName, number>> {
+    if (this.cachedAllActualDecimals) {
+      return this.cachedAllActualDecimals;
+    }
+    const entries = await Promise.all(
+      this.getEvmChains().map(async (chain) => {
+        const token = this.app.router(this.app.getContracts(chain));
+        return [chain, await this.getActualDecimals(chain, token)];
+      }),
+    );
+
+    this.cachedAllActualDecimals = Object.fromEntries(entries);
+
+    return Object.fromEntries(entries);
+  }
+
+  async getActualDecimals(
+    chain: ChainName,
+    hypToken: TokenRouter,
+  ): Promise<number> {
+    const expectedConfig = this.configMap[chain];
+    let decimals: number | undefined = undefined;
+
+    if (isNativeConfig(expectedConfig)) {
+      decimals =
+        this.multiProvider.getChainMetadata(chain).nativeToken?.decimals;
+    } else if (isSyntheticConfig(expectedConfig)) {
+      decimals = await (hypToken as unknown as ERC20).decimals();
+    } else if (isCollateralConfig(expectedConfig)) {
+      const collateralToken = await this.getCollateralToken(chain);
+      decimals = await collateralToken.decimals();
+    }
+
+    if (!decimals) {
+      throw new Error('Actual decimals not found');
+    }
+
+    return decimals;
+  }
+
+  async getCollateralToken(chain: ChainName): Promise<ERC20> {
+    const expectedConfig = this.configMap[chain];
+    let collateralToken: ERC20 | undefined = undefined;
+
+    if (isCollateralConfig(expectedConfig)) {
       const provider = this.multiProvider.getProvider(chain);
-      let collateralToken: ERC20;
 
       if (expectedConfig.type === TokenType.XERC20Lockbox) {
         const collateralTokenAddress = await IXERC20Lockbox__factory.connect(
           expectedConfig.token,
           provider,
         ).callStatic.ERC20();
-        collateralToken = await ERC20__factory.connect(
+        collateralToken = ERC20__factory.connect(
           collateralTokenAddress,
           provider,
         );
       } else {
-        collateralToken = await ERC20__factory.connect(
+        collateralToken = ERC20__factory.connect(
           expectedConfig.token,
           provider,
         );
       }
-      const actualToken = await (
-        hypToken as unknown as HypERC20Collateral
-      ).wrappedToken();
-      if (!eqAddress(collateralToken.address, actualToken)) {
-        const violation: TokenMismatchViolation = {
-          type: 'CollateralTokenMismatch',
-          chain,
-          expected: collateralToken.address,
-          actual: actualToken,
-          tokenAddress: hypToken.address,
-        };
-        this.addViolation(violation);
-      }
+    }
+    if (!collateralToken) {
+      throw new Error('Collateral token not found');
+    }
+    return collateralToken;
+  }
+
+  async checkDecimalConsistency(
```

Comment:
not an async function - will also fix the lint/prettier errors you're getting
```suggestion
  checkDecimalConsistency(
```